### DISCUSSION TITLE
Programmatically get the path for installations

### DISCUSSION THREAD

**Discussion by oldgalileo** [mason-org/mason.nvim#33] [Q&A] [answered]
How can you get the path to the install point for `codelldb` (for example)?

**Comment by williamboman** [comment, marked as answer]
I haven't fully figured out the nicest way this could be done, so the following is subject to change in the future, but for now you can do this:

```lua
local mason_registry = require("mason-registry")
local codelldb = mason_registry.get_package("codelldb") -- note that this will error if you provide a non-existent package name
codelldb:get_install_path() -- returns a string like "/home/user/.local/share/nvim/mason/packages/codelldb"
```

**Reply by przbadu** [reply]
In my case, I have installed `codelldb` using MasonInstall and also if I go to `"/home/user/.local/share/nvim/mason/packages/codelldb"
` path, I can clearly see codelldb is available and still when I open a rust file I see following error:

![image](https://github.com/williamboman/mason.nvim/assets/4189129/fca48e27-b2d6-44a1-9c78-68626d4f16ce)

**Reply by przbadu** [reply]
I have following configuration:

```lua
local rt = require('rust-tools')
local mason_registry = require('mason-registry')

local codelldb = mason_registry.get_package("codelldb") -- note that this will error if you provide a non-existent package name
codelldb:get_install_path() -- returns a string like "/home/user/.local/share/nvim/mason/packages/codelldb"

local extension_path = codelldb:get_install_path() .. "/extension/"
local codelldb_path = extension_path .. "adapter/codelldb"
local liblldb_path = extension_path .. "lldb/lib/liblldb.so"

rt.setup({
  dap = {
    adapter = require("rust-tools.dap").get_codelldb_adapter(codelldb_path, liblldb_path),
  },
  server = {
    on_attach = function(_, bufnr)
      -- Hover actions
      vim.keymap.set("n", "<Leader>k", rt.hover_actions.hover_actions, { buffer = bufnr })
      -- Code action groups
      vim.keymap.set("n", "<Leader>a", rt.code_action_group.code_action_group, { buffer = bufnr })
    end,
  },
  tools = {
    hover_actions = {
      auto_focus = true,
    }
  }
})
```

**Reply by DuyTC1811** [reply]
i also config like you but not working

**Reply by TomzBench** [reply]
I had the same error (Cannot find package codelldb) ..however, it is because I tried to call get_package() before calling mason setup

**Reply by svfat** [reply]
@TomzBench thanks, I am using lazy.nvim and your comment has helped me to solve the issue. Not sure if it is a right approach or not, but it seems I fixed it by calling mason setup in the config
```lua
{
    'williamboman/mason.nvim',
    config = function()
        require("mason").setup({})
    end
},
```

**Reply by fromSmolsoft** [reply]
> @TomzBench ...  I fixed it by calling mason setup in the config
> 
> ```lua
> {
>     'williamboman/mason.nvim',
>     config = function()
>         require("mason").setup({})
>     end
> },
> ```
 
Isn't better to add mason to `dependencies ={...} ` to the plugin that requires it to ensure Mason is loaded when plugin that requires Mason is loaded by Lazy at the time of need?  

```lua
{ "repo/plugin",
    dependencies = {  "williamboman/mason.nvim",  },
}
```

**Reply by Alan-John-Byrne** [reply]
> I haven't fully figured out the nicest way this could be done, so the following is subject to change in the future, but for now you can do this:
> 
> ```lua
> local mason_registry = require("mason-registry")
> local codelldb = mason_registry.get_package("codelldb") -- note that this will error if you provide a non-existent package name
> codelldb:get_install_path() -- returns a string like "/home/user/.local/share/nvim/mason/packages/codelldb"
> ```

Sadly this no longer works. The get_install_path method was completely removed.

**Reply by lpiepiora** [reply]
@Alan-John-Byrne 

> Sadly this no longer works. The get_install_path method was completely removed.

I had the same problem, there is a changelog note: https://github.com/mason-org/mason.nvim/blob/7c7318e8bae7e3536ef6b9e86b9e38e74f2e125e/CHANGELOG.md?plain=1#L65

I'm using `vim.fn.expand "$MASON/packages/codelldb"` for example.

**Reply by Alan-John-Byrne** [reply]
> @Alan-John-Byrne 
> 
> 
> 
> > Sadly this no longer works. The get_install_path method was completely removed.
> 
> 
> 
> I had the same problem, there is a changelog note: https://github.com/mason-org/mason.nvim/blob/7c7318e8bae7e3536ef6b9e86b9e38e74f2e125e/CHANGELOG.md?plain=1#L65
> 
> 
> 
> I'm using `vim.fn.expand "$MASON/packages/codelldb"` for example.
> 
> 

Thanks for the heads up 👍

**Reply by MouadBNL** [reply]
Is there some other api that replaces that ?

**Reply by fromSmolsoft** [reply]
> Is there some other api that replaces that ? 

Like the one in [comment](https://github.com/mason-org/mason.nvim/discussions/33#discussioncomment-13095373) above?

**Reply by Shimoon** [reply]
> Is there some other api that replaces that ?

You can use the `$MASON` environment variable instead, as reported in the [changelog](https://github.com/mason-org/mason.nvim/blob/main/CHANGELOG.md#package-api-changes).

**Reply by qiuxiaomu** [reply]
$MASON is not loaded until neovim is full initialised; if you need it before initialisation (e.g. fetching a particular package's path for another neovim plugin), one way to have it is:
`mason_root = require("mason.settings").current.install_root_dir`
